src/sparseml/transformers/sparsification/trainer.py

```diff
@@ -758,8 +736,13 @@ def check_disable(self, epoch: float, force: bool = False):
             self.disable_amp(epoch)
 
     def qat_active(self, epoch: float) -> bool:
-        return (self.trainer.manager and self.trainer.manager.qat_active(epoch)) or any(
-            bool(man.quantization_modifiers) for man in self.trainer.arch_managers
+        return (
+            self.trainer.manager
+            and self.trainer.manager.qat_active(epoch)
+            or any(
+                bool(stage.quantization_modifiers)
+                for stage in self.trainer.arch_manager.modifiers
+            )
```

Comment:
```suggestion
            or self.trainer.arch_manager.quantization_modifiers
```

quantization_modifiers calls into `iter_modifiers()` under the hood which flattens stages for you!